nlp_primitives/utilities.py

```diff
@@ -7,25 +6,25 @@
 
 
 def clean_tokens(textstr):
+    textstr = textstr.translate(str.maketrans('', '', string.punctuation))
+    textstr = [ch for ch in textstr.split(' ') if len(ch) > 0]
+
     try:
         swords = set(nltk.corpus.stopwords.words('english'))
+        to_remove = set(string.punctuation).union(swords)
+        textstr = [ch.lower() for ch in textstr if ch not in to_remove]
     except LookupError:
         nltk.download('stopwords')
         swords = set(nltk.corpus.stopwords.words('english'))
+        to_remove = set(string.punctuation).union(swords)
+        textstr = [ch.lower() for ch in textstr if ch not in to_remove]
+
+    try:
+        textstr = [wn.lemmatize(w) for w in textstr]
+    except LookupError:
+        nltk.download('wordnet')
+        textstr = [wn.lemmatize(w) for w in textstr]
+
     finally:
-        try:
-            textstr = nltk.tokenize.word_tokenize(textstr)
-        except LookupError:
-            nltk.download('punkt')
-            textstr = nltk.tokenize.word_tokenize(textstr)
-        finally:
-            try:
-                wn.lemmatize("")
-            except LookupError:
-                nltk.download('wordnet')
-            finally:
-                processed = [ch.lower() for ch in textstr if ch not in
-                             set(string.punctuation).union(swords)]
-                processed = ['0' if re.search('[0-9]+', ch) else ch for ch in processed]
-                processed = [wn.lemmatize(w) for w in processed]
-                return processed
+        textstr = ['0' if any(map(str.isdigit, ch)) else ch for ch in textstr]
```

Comment:
similar comment to above (in removing punctuation), just wondering what the reason is for moving away from regex, but I do think this looks very clean!

Reply:
I removed the regex here to add just a little more optimization - the `re.search` runtime is the third largest box in the second cProfile screenshot I posted, it took about 50 seconds. Removing the regex sped things up noticeably, the same operation was cut down to 20 seconds, so `clean_tokens` is down to running in 340 seconds total.

Apparently, the way regex searches through text includes a lot of backtracking and jumping forward, and it can be really inefficient on some types of datasets, depending on what you're searching for (https://towardsdatascience.com/regex-performance-in-python-873bd948a0ea), so I figured it would be safer to pull them out of here at least for now.

Reply:
amazing, makes a lot of sense, cool that you looked into this!